### ISSUE TITLE
Distribution story for the Python media worker

### ISSUE THREAD

**Issue by dyakubu** [dyakubu/scout#11]
The media worker is currently development-only: `main.go` launches it with a hardcoded, repo-relative command (`uv run --project media media/worker.py`), and nothing about the worker ships in the release archive. A user who installs from a release tarball or Homebrew can't turn on image search at all. README already calls this out under Status: "Image search is dev-only; there's no packaged distribution story for the Python media worker yet."

This issue tracks closing that gap, keeping the worker in Python.

## What blocks it

1. **Launch command is repo-relative.** `main.go` hardcodes `[]string{"uv", "run", "--project", "media", "media/worker.py"}`. It needs to resolve against the binary's own directory the way `config.resolveEmbedderPaths` already does for the text model, and probably become a `media.worker_command` config key.
2. **Nothing to point it at.** No Python runtime, no dependencies, no weights in the archive. `media.model_dir` defaults to `""`, and `clip.py` downloads ~605MB from HF on first run.

## The size problem

Measured on darwin/arm64:

| | on disk | compressed |
|---|---|---|
| current release archive (binary + text model + onnxruntime) | — | **36 MB** |
| `media/.venv` (torch + transformers) | **764 MB** | 183 MB |
| ↳ torch alone | 557 MB | — |
| CLIP weights (`pytorch_model.bin`, fp32) | 605 MB | ~560 MB |

Bundling the stack as it exists today means a **~800MB archive per platform**, four platforms per release. torch's wheel is 127MB on macOS arm64 but 555MB on linux-x86_64 from PyPI (179MB from the `+cpu` index) and 454MB on linux-aarch64.

## Prerequisite: swap the worker backend from torch to onnxruntime

torch is 73% of the venv and the reason the weights are 605MB. Running the same model (`openai/clip-vit-base-patch32`) through `onnxruntime` instead keeps `worker.py`, the stdio JSON-lines protocol, and `mediaworker/client.go` unchanged — only `clip.py`'s three functions change.

Validated against the current torch implementation, same 3 images / 4 queries, int8 ONNX towers from `Xenova/clip-vit-base-patch32`:

```
                                   cats.jpg   bear.jpg   food.png
onnxruntime int8   cats query        0.3074     0.1503     0.1498
                   bear query        0.2053     0.2998     0.2004
                   food query        0.2082     0.1994     0.2441
torch reference    cats query        0.2963     0.1267     0.1481
                   bear query        0.1753     0.2987     0.1807
                   food query        0.1991     0.1851     0.2345

cosine(onnx image embed, torch image embed): 0.982, 0.983, 0.955
```

Identical ranking; the residual gap is int8 quantization (bicubic-vs-bilinear resize accounts for ~0.01 of it).

What it buys:

- dependencies: **764MB → 144MB** on disk / 42MB compressed (onnxruntime 77MB, numpy 22MB, PIL 15MB, tokenizers 10MB). `transformers` and `huggingface_hub` drop out entirely.
- weights: **605MB → 154MB** on disk / 113MB compressed for the int8 vision+text pair. (q4f16 is ~126MB, untested; fp32 is 606MB if bit-parity with today is wanted.)
- worker startup: see below.

Costs:

- `CLIPProcessor` goes away — image preprocessing (resize shortest edge 224 bicubic → center crop → rescale → mean/std → CHW) becomes ~10 lines of numpy/Pillow, and tokenization moves to `tokenizers` reading the same `tokenizer.json`. It replicates the processor almost exactly (0.996–1.000 cosine against `CLIPProcessor` output), but nothing errors if it's subtly wrong.
- Export input contracts are frozen and model-specific: CLIP B/32's text tower accepts `input_ids` only and rejects `attention_mask` (`Invalid input name: attention_mask`), and other CLIP-family exports require exactly 77 tokens (trimming padding gives `Attempting to broadcast an axis by a dimension other than 1. 8 by 77`).
- **Bad quantization fails silently.** A MobileCLIP-S0 int8 vision tower tested alongside this returned well-formed 512-float vectors that ranked every query identically — no error, no warning. This needs a golden test: fixed images + queries, assert cosine against a stored reference.
- Embeddings shift by ~0.02–0.05 cosine, so existing `vec_media` rows are invalidated. `mediaModelID` is already the staleness mechanism — bumping it re-embeds them.
- `optimum[onnxruntime]`, which would keep the `from_pretrained`/`CLIPProcessor` API on an ONNX backend, installs torch anyway: that venv measures **847MB**, larger than the current one. There's no middle path.

## Startup

Measured spawn → first result through the real stdio protocol:

| | first image embed | first query embed |
|---|---|---|
| today (`uv run` + torch) | 2.34–2.48s | 2.31s |
| standalone python + onnxruntime | 0.33–0.36s | 0.33s |

Decomposed: `uv run` overhead is ~20ms and interpreter start ~20ms, so essentially all 2.4s is `import torch` plus loading the fp32 checkpoint. **Bundling the runtime without switching backends buys nothing here.**

`Search` (`search/searcher.go:148`) runs `searchFiles` before `searchMedia`, and `EmbedText` blocks until the worker has loaded its model, so ~2.3s sits on the critical path of every media-enabled `scout find`. For `scout index` it's mostly hidden behind the walk and text embedding. (Warm page cache; cold widens the gap — 557MB of torch dylibs vs 77MB of onnxruntime.)

## Distribution options

**A. Bundle a relocatable runtime in the archive.** CI downloads python-build-standalone (27MB macOS / 31MB linux / 25MB windows, stripped) and installs the deps into it with `uv pip install --python`; `scripts/package-release.sh` copies the tree plus `worker.py`/`clip.py` into `media/`. Verified relocatable: built the tree, moved it to a different path, ran the real embedding workload under `python3 -I`, 0.9819 cosine against the torch reference.
- with onnxruntime: **~218MB archive** (36 + 69 + 113), ~430MB installed
- with torch: ~810MB archive
- Homebrew needs no formula change — it does `libexec.install Dir["*"]` and symlinks only the binary, so a `media/` subtree comes along automatically.

**B. Freeze with PyInstaller/Nuitka.** Similar size, single executable instead of an interpreter tree. Hidden-import wrangling, and it fights hard with torch (much less so with onnxruntime). Probably not worth the build complexity over A.

**C. Separate "media pack" release asset + `scout media install`.** Base archive stays 36MB; a second asset (~180MB) holds runtime + model, downloaded on demand into scout's config dir, checksum-verified. Keeps text-only installs small and Homebrew light; costs a new command and a fetch/verify/unpack path. Natural home for the existing "model missing → fetch it" logic in `clip.py`.

**D. Publish the worker to PyPI** (`uv tool install scout-media`), discovered on `PATH`. Cleanest boundary, no bundling — but doesn't put the model next to the binary.

## Proposed order

1. Swap the worker backend to onnxruntime + add the golden test. Independent of everything else, and load-bearing for every option.
2. Make the launch command execDir-relative / config-driven.
3. Ship it: A for one self-contained archive, C if 218MB for text-only users is too much.

## Open questions

- **Weight licensing.** `openai/clip-vit-base-patch32` declares no license on its HF model card (the CLIP *code* is MIT; the weights' terms are unstated). This matters when redistributing in a release archive rather than having the user download it. Needs resolving before A or C.
- Bundled licenses/NOTICE for CPython, onnxruntime, Pillow, numpy, tokenizers.
- macOS Gatekeeper quarantine on a manually downloaded archive containing an interpreter and shared libraries (the archive already ships `libonnxruntime.dylib`, so not new, but a bundled python widens the surface).
- CI time and release asset size across the four-platform matrix.
